### PR TITLE
Add UUID to events, based on Segment's messageId

### PR DESCRIPTION
**Pull request by bretthoerner**
Internal context: https://posthoghelp.zendesk.com/agent/tickets/12556

Segment documentation about `uuidv5.uuidv5` being available: https://segment.com/docs/connections/functions/destination-functions/#runtime-and-dependencies

We are using their `messageId` to generate a UUID to remove duplicates.

---

The repo [says to ping Tim](https://github.com/PostHog/posthog-segment/blob/main/README.md#development) for updates. He (or whoever can) should probably run this against real test data inside of Segment's app. It seems like they have a nice interface in there for testing this out?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ async function onTrack(event, settings) {
     const endpoint = new URL(`${baseUrl}/capture/`)
     endpoint.searchParams.set('ts', event.timestamp)
 
+    const namespace = uuidv5.uuidv5('null', 'PostHog', true)
+    const uuid = uuidv5.uuidv5(namespace, event.messageId)
```
Comment (bretthoerner) [inline index.js:49]: `uuidv5` docs show this example for making a custom namespace: https://www.npmjs.com/package/uuidv5

(I don't think it's worth worrying about whether we can cache in global state or anything like that...)

**Review by tkaemming — APPROVED** [review]
Makes sense to me, assuming our understanding of the available dependency is correct…

**Review by xvello — APPROVED** [review]
LGTM, thanks! We'll have to get a new version reviewed by the Segment folks, I'd love to bundle the other change to add `$geoip_disable` to groupidentify events. They are currently going through person processing due to the geoip `$sets` while it's not needed.

**Comment by bretthoerner** [comment]
> I'd love to bundle the other change to add `$geoip_disable` to groupidentify events. They are currently going through person processing due to the geoip `$sets` while it's not needed.

Can you verify https://github.com/PostHog/posthog-segment/pull/20/commits/99c7c66bdce2d017ad9f302dde711a49c2d938c1 ?

**Pull request merged**